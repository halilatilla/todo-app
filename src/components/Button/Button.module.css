.button {
  @apply h-[42px] rounded-lg border border-gray-300 px-4 text-sm font-medium capitalize hover:bg-gray-600 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 disabled:select-none disabled:opacity-50 disabled:hover:bg-transparent;
}

.icon {
  @apply border-none;
}
