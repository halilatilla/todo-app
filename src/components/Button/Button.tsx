import { FC } from 'react'
import classnames from 'classnames'

interface Props {
  label?: string
  onClick?: () => void
  className?: string
  disabled?: boolean
}

const Button: FC<Props> = ({ label, onClick, className, children, disabled, ...rest }) => {
  return (
    <button
      className={classnames(
        'h-[42px] rounded-lg border border-gray-300 text-sm font-medium shadow-sm hover:bg-gray-700 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 disabled:select-none disabled:opacity-50 disabled:hover:bg-transparent',
        className,
      )}
      onClick={onClick}
      disabled={disabled}
      {...rest}
    >
      {label} {children}
    </button>
  )
}

export default Button
