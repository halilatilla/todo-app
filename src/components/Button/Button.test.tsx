import { render, screen } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
  const DATA = {
    label: '',
    onClick: () => {},
    className: '',
    disabled: false,
  }

  it('renders correctly', () => {
    const { container } = render(<Button {...DATA} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Button', () => {
    render(<Button {...DATA} />)

    const button = screen.getByRole('button')

    expect(button).toBeInTheDocument()
  })
})
