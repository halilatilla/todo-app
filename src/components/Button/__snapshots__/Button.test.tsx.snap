// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="button undefined"
    type="button"
  >
    
     
  </button>
</div>
`;
